packages/components/src/public/files/TemplateDownloadButton.tsx

```diff
@@ -1,155 +1,173 @@
-import React, { FC, memo, useCallback, useEffect, useMemo, useState } from 'react';
+import React, { FC, memo, useCallback, useMemo, useState } from 'react';
+import classNames from 'classnames';
 
 import { PermissionTypes } from '@labkey/api';
 
 import { User } from '../../internal/components/base/models/User';
 import { RequiresPermission } from '../../internal/components/base/Permissions';
 import { DropdownButton, MenuItem } from '../../internal/dropdowns';
 import { ImportTemplate } from '../QueryInfo';
-import { LoadingSpinner } from '../../internal/components/base/LoadingSpinner';
 import { SchemaQuery } from '../SchemaQuery';
 import { useAppContext } from '../../internal/AppContext';
 import { downloadAttachment } from '../../internal/util/utils';
 import { DisableableMenuItem } from '../../internal/components/samples/DisableableMenuItem';
 import { useServerContext } from '../../internal/components/base/ServerContext';
 import { getAppHomeFolderPath } from '../../internal/app/utils';
+import { LoadingState } from '../LoadingState';
+
+const TITLE = 'Download Template';
 
 interface Props {
     className?: string;
     defaultTemplateUrl?: string;
-    dropDownClassName?: string;
     isGridRenderer?: boolean;
-    onDownloadDefault?: () => void;
+    onDownloadDefault?: () => Promise<void>;
     schemaQuery?: SchemaQuery;
     text?: string;
     user?: User;
 }
 
-export const TemplateDownloadButton: FC<Props> = memo(props => {
+const TemplateDownloadButtonImpl: FC<Props> = memo(props => {
     const {
-        schemaQuery,
+        className = 'small-right-spacing',
```

Comment:
This is a nitpick that's kind of unrelated to this PR, but could be addressed in this PR:

1. I believe this `small-right-spacing` class is never needed. It's the class we apply by default, and I think that means it's the class that gets added when rendered in a grid, which is completely unnecessary. Also it's a class that applies padding, when we should be applying margin.
2. We could drop this whole `className` prop if we just got smarter about where we put the `className`. Namely, instead of adding a `className` here, we should add a `className` to the div in `PageDetailHeader` that renders buttons, something like `page-header__buttons` and then add a style that is something like: `.page-header__buttons > * { margin-right: 15px }`.